### ISSUE TITLE
Backend doc-generation PoC.

### ISSUE THREAD

**Issue by iliapolo** [cdklabs/construct-hub#139]
Implement a hacky solution for backend doc generation and deploy to environment so we can validate the affects it has on performance.

**Comment by iliapolo** [comment]
PoC is complete. Updating the findings:

The TLDR is basically that while generating the markdown on the backend removes network bottlenecks, it **does not** manifest as a significant overall performance improvement. 

The reason is that the main bottleneck right now is the pretty rendering of the markdown source itself, which is done using the [\<ReactMarkdown\>](https://github.com/cdklabs/construct-hub-webapp/blob/main/src/components/PackageDocs/Body.tsx#L77) component. 

To support this statement, we ran two performance analysis using `Lighthouse`. One for rendering a pretty markdown, and one for rendering it as raw text. Here are the results:

### Markdown Format

<img width="975" alt="Screen Shot 2021-06-30 at 2 26 42 PM" src="https://user-images.githubusercontent.com/1428812/123960115-3795cb00-d9af-11eb-9991-f24a44cc1bca.png">

> Full report here: [performance-report-eks-markdown.txt](https://github.com/cdklabs/construct-hub/files/6740864/performance-report-eks-markdown.txt) (Change .txt to .html)

### Raw Format

<img width="959" alt="Screen Shot 2021-06-30 at 2 33 35 PM" src="https://user-images.githubusercontent.com/1428812/123960971-2bf6d400-d9b0-11eb-860f-00a5636ffc7d.png">

> Full report here: [performance-report-eks-raw.txt](https://github.com/cdklabs/construct-hub/files/6740871/performance-report-eks-raw.txt) (Change .txt to .html)

----

We can see that *Time to interactive* in the raw case is `0.9s`, while in the markdown case its `3.6s`. 
Both are actually not so bad, and are lower than 5s, which is our current goal. 

Even in light of these findings, we are proceeding with the backend migration, for the following reasons:

1. It's the right thing to do, architecturally. This content is static and should be generated once at ingestion time, regenerating it on every request is wasteful, and may cause more performance issues down the road.
2. It does provide a substantial improvement for slower internet connection environments (such as 3G mobile).
3. It makes the front-end much less complicated.
5. It makes it so we don't have to index all versions of all packages to create documentation (we can just `npm` install the package on the backend).
6. It allows running advanced sanitation logic on external inputs (user defined README).
7. We will have to do some backend pre-processing regardless in order to support packages with a large number of submodules (aws-cdk v2) - so with not a lot of effort, we can go all the way here.
8. It allows for external consumers to consume the documentation page and embed it in their own site. (aws docs maybe?)

Closing this out.
